instrumentation/opentelemetry-instrumentation-django/tests/test_middleware.py

```diff
@@ -430,3 +433,67 @@ def test_tracer_provider_traced(self):
         self.assertEqual(
             span.resource.attributes["resource-key"], "resource-value"
         )
+
+class TestDjangoWithOtherFramework(SimpleTestCase, TestBase, WsgiTestBase):
+    @classmethod
+    def setUpClass(cls):
+        conf.settings.configure(ROOT_URLCONF=modules[__name__], WSGI_APPLICATION="tests.wsgi.application")
+        super().setUpClass()
+
+    def setUp(self):
+        super().setUp()
+        setup_test_environment()
+        
+        # conf.settings.WSGI_APPLICATION = "wsgi.application"
+        # application = get_internal_wsgi_application()
+        # application = get_wsgi_application()
+        # _DjangoMiddleware
+        # self.application = OpenTelemetryMiddleware(application)
+        # _django_instrumentor.instrument()
+        # self.application = OpenTelemetryMiddleware(application, tracer_provider=self.tracer_provider)
+        # conf.settings.configure(ROOT_URLCONF=modules[__name__], WSGI_APPLICATION="application")
+        # conf.settings.WSGI_APPLICATION="application"
+        
+
+    def tearDown(self) -> None:
+        super().tearDown()
+        teardown_test_environment()
+        _django_instrumentor.uninstrument()
+    
+    @classmethod
+    def tearDownClass(cls):
+        super().tearDownClass()
+        conf.settings = conf.LazySettings()
+
+
+    def test_with_another_framework(self):
+        environ = RequestFactory()._base_environ(
+            PATH_INFO="/span_name/1234/",
+            CONTENT_TYPE="text/html; charset=utf-8",
+            REQUEST_METHOD="GET"
+        )
+        response_data = {}
+        def start_response(status, headers):
+            response_data["status"] = status
+            response_data["headers"] = headers
+        
+        result = self.create_tracer_provider()
+        tracer_provider, exporter = result
+        self.exporter = exporter
+        
+        
+        _django_instrumentor.instrument(tracer_provider=tracer_provider)
+        application = get_internal_wsgi_application()
+        application = OpenTelemetryMiddleware(application, tracer_provider=tracer_provider)
+        resp = application(environ, start_response)
+        
+        # resp = Client().get("/span_name/1234/")
+        # self.assertEqual(200, resp.status_code)
+
+        # span_list = self.memory_exporter.get_finished_spans()
+        span_list = self.exporter.get_finished_spans()
+        # print(span_list)
+        self.assertEqual(trace.SpanKind.INTERNAL, span_list[0].kind)
+
+        #Below line give me error "index out of the range" as there is only one span created where it should be 2.
```

Comment:
It appears that your server span from middleware started but never ended. Please try changing the line 488 to `next(application(environ, start_response))` and check.

Reply:
Sure. Done and changes are pushed. Please do review them

Reply:
I only intend to suggest that hack to validate the hypothesis. You should use test client instead.

Reply:
With Client() it was not creating the WSGI span. That is the reason I have to go this way